### DISCUSSION TITLE
Varied results while applying a watermark on a image using IM on docker container

### DISCUSSION THREAD

**Discussion by dekay14** [ImageMagick/ImageMagick#2021] [General] [answered]
Hello,
I am pretty new to ImageMagick and am trying to figure out how to apply a watermark on an image. When I run the following convert command on my mac (High Sierra). I see the desired results, both on command line and programatically (I am using java Process Builder).

`convert logo.gif \
\( -background none -pointsize 24 -fill red label:copyright -rotate 20 -write mpr:tile +delete \) \
\( +clone -tile mpr:tile -draw "color 0,0 reset" \) \
-compose over -composite wm_cross_logo.gif`

But when I run the same command on a docker container where ImageMagick is installed and the app is running, I see that the actual image is missing and the watermark is all blurred.
I see the following message in the console logs:

Fontconfig warning: ignoring C.UTF-8: not a valid language tag

Version instllaed on Mac: ImageMagick 7.0.10-11
version installed on docker container: Version: ImageMagick 7.0.5-10

![wm_cross_logo](https://user-images.githubusercontent.com/6627708/81991700-b9cdff00-95f6-11ea-93b5-85d4f286c33f.gif)

Any direction is appreciated.

Thanks!

**Comment by fmw42** [comment, marked as answer]
There is something odd going on. The image shows up more properly in my email, but not here in the forum. Nevertheless, use magick in IM 7, not convert. Try this with magick in place of convert and no period at the end after .gif

`magick logo: \( -background none -pointsize 24 -fill red label:copyright -rotate 20 -write mpr:tile +delete \) \( +clone -tile mpr:tile -draw "color 0,0 reset" \) -compose over -composite wm_cross_logo.gif
`

I get 


![wm_cross_logo](https://user-images.githubusercontent.com/13458218/81994170-1aac0600-95fc-11ea-8e18-139b3b01cfba.gif)

**Reply by dekay14** [reply]
magick or convert commands, I get the same result where the actual image is lost and the text is blurred. I feel it's something about the IM installed on the Docker container.

**Reply by fmw42** [reply]
The docker version of IM is very old. Perhaps it had a bug. Can you upgrade it? But first try saving as PNG or TIFF rather than GIF

**Reply by dekay14** [reply]
I get the same result even with tiff and png. I always seem to get the error message after I run the command, but it doesn't seem like is causing the issue.

`Fontconfig warning: ignoring C.UTF-8: not a valid language tag.
`
I will updating the IM version. Thank you.

**Reply by fmw42** [reply]
The error message seems to be associated with the fontconf delegate. Do you have that installed?  What do you get from 

`magick -version`

Please show all that is returned.  If it is installed, then perhaps it needs upgrading. Be sure to use current versions of all your delegate libraries.

**Reply by dekay14** [reply]
Please see below:

magick -version
Version: ImageMagick 7.0.5-10 Q16 x86_64 2017-06-05 http://www.imagemagick.org
Copyright: © 1999-2017 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Features: Cipher HDRI Modules 
Delegates (built-in): fontconfig freetype gslib jng jpeg lcms ltdl png ps tiff webp xml zlib

**Reply by fmw42** [reply]
Looks OK. But IM is very old and perhaps your delegates are old also.

**Reply by dekay14** [reply]
That was it!, i updated the base image which in trun update IM to a new version. I am able to generate the watermarked image now. Thank you!

**Reply by narendrareddy527** [reply]
@dekay14  Can you please share how to update base image with ImageMagick? I want to know how to use ImageMagick in docker
